### ISSUE TITLE
Shadow Tomography

### ISSUE THREAD

**Issue by Zoufalc** [Qiskit/qiskit#8581]
### What should we add?

It would be great if we had a shadow tomography module - which enables the extraction of particular information about a quantum state with less measurements than full state tomography.

References:
[Shadow Tomography of Quantum States](https://arxiv.org/abs/1711.01053)

**Comment by ShellyGarion** [comment]
I agree that shadow tomography would be a great contribution to Qiskit. The main question is where does it belong?

Currently, the tomography experiments belong to Qiskit Experiments: https://qiskit.org/documentation/experiments/
including:
- [StateTomography](https://qiskit.org/documentation/experiments/stubs/qiskit_experiments.library.tomography.StateTomography.html#qiskit_experiments.library.tomography.StateTomography)
- [ProcessTomography](https://qiskit.org/documentation/experiments/stubs/qiskit_experiments.library.tomography.ProcessTomography.html#qiskit_experiments.library.tomography.ProcessTomography)
- PR for gate set tomography:  https://github.com/Qiskit/qiskit-experiments/pull/503

Another possibility may be Qiskit prototypes: https://github.com/qiskit-community/prototypes

**Comment by chriseclectic** [comment]
Shadow tomography is just partial linear inversion state tomography in the Pauli basis with randomly sampled basis measurements. The only difference is you don't reconstruct the density matrix first, but just store the "state" as a list of measurement probs and corresponding stabilizer basis states that were measured. Hence basically everything exists already in qiskit experiments to do this except for having a nice representation of the "shadow state" to return that you can use to compute quantities from it.

One approach could be to add a new `QuantumState` subclass like `ShadowState` that is basically a density matrix stored as a linear combination of stabilizer states so you can have an interface to computing quantities like expectation values and fidelity's that you might want to use these for.

The biggest caveat is that the estimated state generally wont have trace = 1 unless you rescale your sampled measurement outcomes (which iirc you typically don't do in shadow estimation protocols as they bias your estimates of expectation values), and it wont correspond to a PSD state (and you can't rescale it to be a PSD state without constructing the dense density matrix first to do eigenvalue rescaling, but if you do this you are exactly doing standard partial linear inversion tomography so don't need anything else than what we already have.)

**Comment by jakelishman** [comment]
I'm going to close this as "wontfix" generally for practical/scope reasons.  This doesn't preclude future discussion, but there's no plan to take this forward any time soon.
